### PR TITLE
Issue #320 Add --follow option to 'minishift logs' command

### LINKED ISSUE #320: Add --follow option to `minishift logs` subcommand
Currently we can't follow up logs of `origin` container but only able to dump it to `stdout`. Docker provide a way to follow running container logs using `--follow` and it would be good UX if we can have it for minishift.

### PR DESCRIPTION
**Pull request by budhrg**
Fix #320 

@hferentschik This work functionally IMO. Also, others have verified it.

### CODE REVIEW

File: pkg/minikube/cluster/cluster.go [hunk 1]
```diff
@@ -43,11 +43,13 @@ import (
 	"github.com/minishift/minishift/pkg/util"
 	minishiftOs "github.com/minishift/minishift/pkg/util/os"
 	"github.com/minishift/minishift/pkg/util/os/atexit"
+	"github.com/pkg/errors"
 	pb "gopkg.in/cheggaaa/pb.v1"
 )
 
 var (
 	logsCmd             = "docker logs origin"
+	logsCmdFollow       = "docker logs origin -f"
```
Comment (gbraad) [inline pkg/minikube/cluster/cluster.go]: Not a big problem, but reusing `logsCmd` makes sense

File: cmd/minishift/cmd/logs.go [hunk 2]
```diff
@@ -44,5 +48,6 @@ var logsCmd = &cobra.Command{
 }
 
 func init() {
+	logsCmd.Flags().BoolVarP(&follow, "follow", "f", false, "Continously print the logs entries")
```
Comment (gbraad) [inline cmd/minishift/cmd/logs.go]: +1 `-f` ;-)

File: pkg/minikube/cluster/cluster.go (hunk 1 above)
Comment (budhrg) [inline pkg/minikube/cluster/cluster.go]: Done :+1:

File: cmd/minishift/cmd/logs.go (hunk 2 above)
Comment (budhrg) [inline cmd/minishift/cmd/logs.go]: @gbraad , that's not required as cobra take care of it  

```
$ minishift logs -h
Gets the logs of the running OpenShift cluster. The logs do not contain information about your application code.

Usage:
  minishift logs [flags]

Flags:
  -f, --follow   Continously print the logs entries
  -h, --help     help for logs
```

Comment (gbraad) [inline cmd/minishift/cmd/logs.go]: I meant, it is good to have the shorthand alias.

File: pkg/minikube/cluster/cluster.go [hunk 3]
```diff
@@ -468,16 +470,32 @@ func GetHostDockerEnv(api libmachine.API) (map[string]string, error) {
...
 // GetHostLogs gets the openshift logs of the host VM.
-func GetHostLogs(api libmachine.API) (string, error) {
+// If follow is specified, it will tail the logs
+func GetHostLogs(api libmachine.API, follow bool) (string, error) {
 	host, err := CheckIfApiExistsAndLoad(api)
 	if err != nil {
 		return "", err
 	}
+
+	if follow {
```
Comment (hferentschik) [inline pkg/minikube/cluster/cluster.go]: Can we not pull this into its own method and make the switch on the _follow_ flag more apparent?  Really we have an is/else here, right?

File: pkg/minikube/cluster/cluster_test.go [hunk 4]
```diff
@@ -50,3 +53,52 @@ func TestLocalBoot2DockerURL(t *testing.T) {
...
+		},
+		{
+			description: "logs -f",
+			follow:      true,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.description, func(t *testing.T) {
+			t.Parallel()
```
Comment (hferentschik) [inline pkg/minikube/cluster/cluster_test.go]: Why parallel. It does not serve any purpose here and makes it harder to to understand what you are testing.

File: pkg/minikube/cluster/cluster_test.go [hunk 5]
```diff
@@ -50,3 +53,52 @@ func TestLocalBoot2DockerURL(t *testing.T) {
 		t.Fatalf("Expected URL : %s", localISOUrl)
 	}
 }
+
+func TestHostGetLogs(t *testing.T) {
```
Comment (hferentschik) [inline pkg/minikube/cluster/cluster_test.go]: I think we discussed this before, I am not so sure of the usefulness of the test. At the very least we should make explicit what this actually does. It verifies that the '-f' flag if specified gets passed to the VM (via ssh). This is by no means a test to verify whether streaming the logs actually works or not.

**Comment by budhrg** [comment]
@hferentschik  Updated the PR as per our discussion.

**Comment by hferentschik** [comment]
Merged, thanks.

**Comment by hferentschik** [comment]
Merged via pull request #1061

**Pull request closed without merging**